service/controller/v2/arm_templates/vmss.json

```diff
@@ -283,7 +281,7 @@
             },
             "properties": {
                 "roleDefinitionId": "[variables('contributorRoleDefinitionId')]",
-                "principalId": "[reference(concat(resourceId('Microsoft.Compute/virtualMachineScaleSets/', parameters('vmssName')),'/providers/Microsoft.ManagedIdentity/Identities/default'),'2015-08-31-PREVIEW').principalId]",
+                "principalId": "[if(parameters('vmssMSIEnabled'), reference(concat(resourceId('Microsoft.Compute/virtualMachineScaleSets/', parameters('vmssName')),'/providers/Microsoft.ManagedIdentity/Identities/default'),'2015-08-31-PREVIEW').principalId, '')]",
```

Comment:
Is it really needed? We have `"condition"` here, so it shouldn't be executed at all.

Reply:
Oh yeeeah, it still tries to execute `reference` even is resource is disabled :/